### ISSUE TITLE
AsyncResult.catch don't work as expected

### ISSUE THREAD

**Issue by AbrahamAlcaina** [swlaschin/DomainModelingMadeFunctional#8]
Hello first, a really good book  👍 congrats.
Here my doubt. 
I try to capture an exception in `asyncResult` with `AsyncResult.catch` but It doesn't work as expected. Here my tests:

Async example: 
```f#
[<Fact>]
let asyncExceptionTest() = 
  let r = async {
              failwith "Ohh nooo!!!"
              return "ok"
            }
          |> Async.Catch
          |> Async.RunSynchronously
  match r with 
  | Choice1Of2 a -> printfn "ok --> %A" a        
  | Choice2Of2 e -> printfn "error -->%A" e
```
This example works well. It captures the exception. Now the same example with `asyncResult`.
```f#
[<Fact>]
let asyncResultTest() = 
  let r = asyncResult {
              failwith "Ohh nooo!!!"
              return "ok"
            }
          |> AsyncResult.catch ( fun ex -> printfn "%A" ex; ex)
          |> Async.RunSynchronously
  match r with 
  | Ok a -> printfn "ok --> %A" a        
  | Result.Error e -> printfn "error -->%A" e    
```
Here the code catch didn't catch the exception. 
Any ideas why?

**Comment by allentiak** [comment]
Just curious... Is this one fixed by #10?

**Comment by AbrahamAlcaina** [comment]
> Just curious... Is this one fixed by #10?

Nop

**Comment by allentiak** [comment]
@AbrahamAlcaina Thanks for your reply :-)

**Comment by swlaschin** [comment]
Sorry for the delay. I finally got round to fixing this! Fixed in fcfe97f06308b6b8d1d7af97761874cf55ac4894
